internal/client/cluster/iam_policy/iam_resource.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package iamclusterclient
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/client/transport"
+	clustermodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/cluster"
+	iammodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/iam_policy"
+	clusteriammodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/iam_policy/cluster"
+)
+
+// New creates a new cluster IAM policy resource service API client.
+func New(transport *transport.Client) ClientService {
+	return &Client{Client: transport}
+}
+
+/*
+Client for cluster IAM policy resource service API.
+*/
+type Client struct {
+	*transport.Client
+}
+
+// ClientService is the interface for Client methods.
+type ClientService interface {
+	ManageV1alpha1ClusterIAMPolicyGet(fn *clustermodel.VmwareTanzuManageV1alpha1ClusterFullName) (*clusteriammodel.VmwareTanzuManageV1alpha1ClusterGetClusterIAMPolicyResponse, error)
+
+	ManageV1alpha1ClusterIAMPolicyPatch(request *clusteriammodel.VmwareTanzuManageV1alpha1ClusterPatchClusterIAMPolicyRequest) (*clusteriammodel.VmwareTanzuManageV1alpha1ClusterPatchClusterIAMPolicyResponse, error)
+
+	ManageV1alpha1ClusterIAMPolicyUpdate(fn *clustermodel.VmwareTanzuManageV1alpha1ClusterFullName, request *iammodel.VmwareTanzuCoreV1alpha1PolicyIAMPolicy) (*clusteriammodel.VmwareTanzuManageV1alpha1ClusterUpdateClusterIAMPolicyResponse, error)
+}
+
+/*
+  ManageV1alpha1ClusterIAMPolicyGet gets policy for a cluster.
+*/
+
+func (c *Client) ManageV1alpha1ClusterIAMPolicyGet(fn *clustermodel.VmwareTanzuManageV1alpha1ClusterFullName) (*clusteriammodel.VmwareTanzuManageV1alpha1ClusterGetClusterIAMPolicyResponse, error) {
+	queryParams := url.Values{}
+
+	if fn.ManagementClusterName != "" {
+		queryParams["fullName.managementClusterName"] = []string{fn.ManagementClusterName}
+	}
+
+	if fn.ProvisionerName != "" {
+		queryParams["fullName.provisionerName"] = []string{fn.ProvisionerName}
+	}
+
+	requestURL := fmt.Sprintf("%s/%s?%s", "v1alpha1/clusters:iam", fn.Name, queryParams.Encode())
```

Comment:
"v1alpha1/clusters:iam" this can be made into a constant and referred in all the places

Reply:
Made constants as `paths` and replaced in all client files.